### ISSUE TITLE
Compile PyMixtComp produces a .dylib file and not .os

### ISSUE THREAD

**Issue by MostafaAbdelrashied** [modal-inria/MixtComp#22]
After compiling, pyMixtCompBridge library can not be found at the following location: build/lib/pyMixtCompBridge.so as mentioned. However, another file build/lib/pyMixtCompBridge .dylib is created. this file can't be read by the python scripts. 

It was created on
1) MacBook Intel-Chip
2) boost==1.8.0
3) boost-python3 (latest version)
4) python==3.10.x

**Comment by Quentin62** [comment]
It's because, you have a mac. I have not tested the python version on a mac.

Have you try copying the .dylib file instead of the .so file in the build.sh script?
https://github.com/modal-inria/MixtComp/blob/staging/pyMixtComp/python/build.sh

If it's working, I will update the build.sh script

**Comment by MostafaAbdelrashied** [comment]
copying the file didn't do anything. Reading a .dylib file in python is not possible in python.
https://stackoverflow.com/questions/2488016/how-to-make-python-load-dylib-on-osx

Here are my trials to solve the issue: (None has worked)

1) I tried copying the .dylib and running the script, but it didn't work. 
`ImportError: cannot import name 'pyMixtCompBridge' from partially initialized module 'pyMixtComp' (most likely due to a circular import) (/Users/56946/workspace/MixtComp/pyMixtComp/python/pyMixtComp/__init__.py)`

2) I tried copying the .dylib and renaming it to .so file. but it didn't work as well.
`ImportError: dlopen(/Users/56946/workspace/MixtComp/pyMixtComp/python/pyMixtComp/pyMixtCompBridge.so, 0x0002): symbol not found in flat namespace (_PyCMethod_New)`

3) I tried running the compiler on Ubuntu to get a .so file as you described In the documentation. Then I tried to use the .so file on my Mac, but it didn't work either. 
`ImportError: dlopen(/Users/56946/workspace/MixtComp/pyMixtComp/python/pyMixtComp/pyMixtCompBridge.so, 0x0002): tried: '/Users/56946/workspace/MixtComp/pyMixtComp/python/pyMixtComp/pyMixtCompBridge.so' (not a mach-o file)`
